Xamarin.Forms.Core/IndicatorsView.cs

```diff
@@ -0,0 +1,285 @@
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	public enum IndicatorShape
+	{
+		Circle,
+		Square
+	}
+
+	public class IndicatorsView : StackLayout
+	{
+		public static readonly BindableProperty PositionProperty = BindableProperty.Create(nameof(Position), typeof(int), typeof(IndicatorsView), default(int), BindingMode.TwoWay, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty CountProperty = BindableProperty.Create(nameof(Count), typeof(int), typeof(IndicatorsView), default(int), propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsCount((int)oldValue, (int)newValue);
+		});
+
+		public static readonly BindableProperty MaxVisibleCountProperty = BindableProperty.Create(nameof(MaxVisibleCount), typeof(int), typeof(IndicatorsView), int.MaxValue, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty IndicatorTemplateProperty = BindableProperty.Create(nameof(IndicatorTemplate), typeof(DataTemplate), typeof(IndicatorsView), default(DataTemplate), propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty IndicatorShapeProperty = BindableProperty.Create(nameof(IndicatorShape), typeof(IndicatorShape), typeof(IndicatorsView), default(IndicatorShape), propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty IndicatorsColorProperty = BindableProperty.Create(nameof(IndicatorsColor), typeof(Color), typeof(IndicatorsView), Color.Silver, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty SelectedIndicatorColorProperty = BindableProperty.Create(nameof(SelectedIndicatorColor), typeof(Color), typeof(IndicatorsView), Color.Gray, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		public static readonly BindableProperty IndicatorsSizeProperty = BindableProperty.Create(nameof(IndicatorsSize), typeof(double), typeof(IndicatorsView), 10.0, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetIndicatorsStyles();
+		});
+
+		internal static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(IndicatorsView), null, propertyChanged: (bindable, oldValue, newValue) =>
+		{
+			(bindable as IndicatorsView)?.ResetItemsSource(oldValue as IEnumerable);
+		});
+
+		static IndicatorsView()
+		{
+		}
+
+		public IndicatorsView()
+		{
+			Orientation = StackOrientation.Horizontal;
```

Comment:
this won't play nice with styling, but anyway, we should drop the SL inheritance

Reply:
@StephaneDelcroix yeah, i thought about it. We could add protected virtual method for handling default SL orientation value.
But as you mentioned, we are refusing SL inheritance at all